### DISCUSSION TITLE
proposal: Make openraft running on stable rust

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/openraft#539] [Ideas]
Hi, OpenRaft pinning build rust version on very old nightly: `nightly-2022-03-30`. Since `1.63` has been released, maybe it's time for use to make openraft running on stable rust?

- openraft only depends on the following features: `backtrace`, `test`
  - `backtrace` has been stabilized. We can remove this feature directly.
  - `test` is still nightly, but seems no code really depend on this (Please correct me)

For a quick demo, I have openraft built on stable rust  (v1.63) locally:

```rust
:( cargo c
    Checking openraft v0.6.4 (/home/xuanwo/Code/datafuselabs/openraft/openraft)
... (ignored all warnings)
warning: `openraft` (lib) generated 22 warnings
    Checking rocksstore v0.1.0 (/home/xuanwo/Code/datafuselabs/openraft/rocksstore)
    Checking sledstore v0.1.0 (/home/xuanwo/Code/datafuselabs/openraft/sledstore)
    Checking memstore v0.2.0 (/home/xuanwo/Code/datafuselabs/openraft/memstore)
    Finished dev [unoptimized + debuginfo] target(s) in 7.00s
cargo c  7.55s user 0.22s system 110% cpu 7.035 total
```

We can address all the warning and update some CI so that everything works as before.

**Comment by drmingdrmer** [comment]
`std::backtrace` is stabilized in rust stable-1.63. But `std::error::Error::backtrace()` is still marked as an unstable feature in rust stable-1.63:

```rust
    #[unstable(feature = "backtrace", issue = "53487")]
    fn backtrace(&self) -> Option<&Backtrace> {
        None
    }
```

It looks like feature `bt` must be kept:(
